code/datums/traits/good.dm

```diff
@@ -188,3 +188,18 @@
 /datum/quirk/bloodpressure/remove()
 	var/mob/living/M = quirk_holder
 	M.blood_ratio = 1
+
+/datum/quirk/night_vision
+	name = "Night Vision"
+	desc = "You can see slightly more clearly in full darkness than most people."
+	value = 1
+	mob_trait = TRAIT_NIGHT_VISION
+	gain_text = "<span class='notice'>The shadows seem a little less dark.</span>"
+	lose_text = "<span class='danger'>Everything seems a little darker.</span>"
+
+/datum/quirk/night_vision/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/obj/item/organ/eyes/eyes = H.getorgan(/obj/item/organ/eyes)
+	if(!eyes || eyes.lighting_alpha)
+		return
+	eyes.Insert(H) //refresh their eyesight and vision
```

Comment:
What the hell? Just modify the eyes' `lighting_alpha` and `see_in_dark` right here and call `H.update_sight()` instead.

Reply:
yeah i just copied the old code :joy: didn't know that was a thing not my code lmao one sec